### PR TITLE
Document clock drift parameters in guide

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #3139 

## Description

Adds a new "Handling Clock Drift" section to the Hermes guide. The docs still need some more context on the issues that may arise when the clock drift parameters are mis-configured. @ancazamfir is doing some testing on this, and we'll add those details to this section when she has done this. 

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: guide/src/advanced/troubleshooting/clock-drift.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Handling Clock Drift
+
+When it comes to distributed systems, clock drift is always a consideration that needs to be 
+taken into account. There is always some amount of drift between the clock of the source chain 
+and the clock of the destination chain that must be accounted for. As far as Hermes is concerned,
+there are two configuration parameters that exist to specify how much clock drift ought to be tolerated: 
+ - `clock_drift` is a correction parameter that specifies how far in the future or past a chain's clock may
+   drift in relation to this client. The destination chain for this client uses this parameter when deciding
+	 whether to accept or reject a new header from the source chain for this client.
```
Comment (ancazamfir) [inline guide/src/advanced/troubleshooting/clock-drift.md:9]:
```suggestion
IBC light client security model requires that timestamp of a header included in client updates for some `client` is within `[now - client.trusting_period, now + client.max_clock_drift)`.

The `trusting_period` is a parameter that is configurable under the `[[chains]]` section and its default value is two thirds of the chain `unbonding_period`.
The rest of this section describes the configuration parameters that determine the `max_clock_drift`.

IBC light client security model requires that the clocks of the reference and host chains are roughly synchronized. Hermes uses the `clock_drift` and `max_block_time` configuration parameters to determine how much clock drift is tolerable between the reference and host chains.
 - `clock_drift` is a correction parameter that specifies how far in the future or past a chain's clock may be from the current time..
```

File: guide/src/advanced/troubleshooting/clock-drift.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ - `max_block_time` is the maximum amount of time that can occur before a new block is created on the chain. 
+
+> **Note:** When it comes to Cosmos SDK chains, a good approximation for `max_block_time` is
+`timeout_propose` + `timeout_commit`, which together sum up to the total amount of time before
+the proposal and commitment steps for a new block time out.
```
Comment (ancazamfir) [inline guide/src/advanced/troubleshooting/clock-drift.md:14]:
```suggestion
> **Note:** For Cosmos SDK chains, a good approximation for `max_block_time` is
`C * (timeout_propose + timeout_commit)`. To allow for some variance in block times while still detecting forward lunatic attacks (todo - add reference), it is recommended to set `C` to be in the `3..5` range. Hermes uses the default value of `30s` which is a good approximation for most Cosmos SDK chains that have 6-10sec block times.
```

File: guide/src/advanced/troubleshooting/cross-comp-config.md [hunk 3]
```diff
@@ -279,3 +279,21 @@ that it needs in order to relay on behalf of a consumer chain.
 
 ### Fix
 Set `ccv_consumer_chain = true` in `config.toml`.
+
+## Mishandling Clock Drift
```
Comment (ancazamfir) [inline guide/src/advanced/troubleshooting/cross-comp-config.md:283]: This seems to be covered already in https://github.com/informalsystems/hermes/blob/master/guide/src/advanced/troubleshooting/cross-comp-config.md#maximum-block-time

**Review by ancazamfir — CHANGES_REQUESTED** [review]
Left some comments, most of them editorial. We need to find a reference to the FLA attacks.  @josef-widder anything in the spec that talks about this?

File: guide/src/advanced/troubleshooting/clock-drift.md (hunk 2 above)
Comment (ancazamfir) [reply guide/src/advanced/troubleshooting/clock-drift.md:14]: It's a constant to allow for variation in block times, mainly due to Tx execution time which is outside of consensus params.

**Comment by josef-widder** [comment]
> Left some comments, most of them editorial. We need to find a reference to the FLA attacks. @josef-widder anything in the spec that talks about this?

What does FLA stand for?

**Comment by seanchen1991** [comment]
Thanks @josef-widder! I think the ADRs are sufficient 🙂

**Review by ancazamfir — APPROVED** [review]
Great! Thanks @seanchen1991

File: guide/src/advanced/troubleshooting/clock-drift.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Handling Clock Drift
+
+IBC light client security model requires that the timestamp of a header included in client updates for some `client` is within `[now - client.trusting_period, now + client.max_clock_drift)`.
```
Comment (DaviRain-Su) [inline guide/src/advanced/troubleshooting/clock-drift.md:3]: should ]

Comment (seanchen1991) [reply guide/src/advanced/troubleshooting/clock-drift.md:3]: Hi @DaviRain-Su, I don't believe this is a typo. The `)` on the right-hand side of the range indicates that the max value of the range is exclusive such that it doesn't include `now + client.max_clock_drift`.

Comment (DaviRain-Su) [reply guide/src/advanced/troubleshooting/clock-drift.md:3]: oh thx your explain, I missunderstand.

**Pull request merged**